internal/kms/vault_sa.go

```diff
@@ -298,6 +293,15 @@ func (kms *vaultTenantSA) getToken() (string, error) {
 			kms.tenantSAName, err)
 	}
 
+	// From Kubernetes v1.24+, secret for service account tokens are not
+	// automatically created. Trying to fetch tokens from service account secret references will fail
+	// refer: https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.24.md \
+	// #no-really-you-must-read-this-before-you-upgrade-1.
+	token, err := kms.createToken(sa, c)
+	if err == nil {
+		return token, nil
+	}
```

Comment:
```suggestion
	token, createTokenErr := kms.createToken(sa, c)
	if createTokenErr == nil {
		return token, nil
	}
```

Reply:
not sure what the change above will cover. If we want to be more specific. Continue to fetch token from the SA only if kubernetes doesn't support TokeRequest API else return all other errors.

Reply:
apologize, the other part of the requested changes was not posted by mistake.

Reply:
Great catch! Yes it makes sense to include the error as part of the final error. I updated the code to include it in the error being returned.